pleiades/vocabularies/vocabularies.py

```diff
@@ -0,0 +1,39 @@
+from zope.component import getUtility
+from zope.interface import directlyProvides
+from zope.schema.interfaces import IVocabularyFactory
+from zope.schema.vocabulary import SimpleTerm
+from zope.schema.vocabulary import SimpleVocabulary
+
+from plone.registry.interfaces import IRegistry
+
+from pleiades.vocabularies.interfaces import IPleiadesSettings
+
+
+def registry_vocabulary(field, value_filter=None):
+    path = field.interface.__identifier__ + '.' + field.__name__
+
+    def vocabulary_factory(context):
+        registry = getUtility(IRegistry)
+        values = registry.get(path, [])
+        values.sort()
+        terms = []
+        for value in values:
+            terms.append(SimpleTerm(
+                value=value['id'],
+                token=value['id'].encode('raw_unicode_escape'),
```

Comment:
raw_unicode_escape should produce 7-bit ascii, with other characters encoded in python literal notation